.gitignore

```diff
@@ -22,3 +22,17 @@ dist-ssr
 *.njsproj
 *.sln
 *.sw?
+
+
+# Compiled JavaScript files
+*.js
+
+# TypeScript declaration files
+*.d.ts
+
+# TypeScript configuration files
+tsconfig.json
```

Comment:
These configuration files need to be committed! They tell the TypeScript compiler what rules they need to enforce on your TypeScript files, and without them my IDE is blowing up with red squiggles.

Reply:
Specifically `tsconfig.json` and `tsconfig.node.json`. I think `vite.config.json` should also be committed. Generally these config files should be shared by your whole team so everyone's IDE works the same.